src/main/java/core/basesyntax/dao/StoreDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import java.util.HashMap;
+import java.util.Map;
+
+public class StoreDaoImpl implements StoreDao {

```

Comment:
If you want to go hard way using DAO please make it properly, first you don`t need to pass Dao into method parameter you should make it field and initialize via constructor. But it will be better if you will interact with storage directly in this task, thus DAO will not be needed